jvcl/run/JvInterpreterParser.pas

```diff
@@ -658,10 +666,28 @@ function TJvInterpreterParser.Token: string;
   if CharInSet(P[0], StConstSymbols10) then
   { number }
   begin
-    while CharInSet(P[0], StConstSymbols10) or (P[0] = '.') do
+    while CharInSet(P[0], StConstSymbols10e) do  
     begin
-      if (P[0] = '.') and (P[1] = '.') then
-        Break;
+      if P[0] = '.' then
+      begin
+        if point_occur //lht��radix point can occur zero or one time
+            or not CharInSet(P[-1],StConstSymbols10) //lht��a radix point must behind a number
```

Comment:
I'm pretty sure the coding style requires the "or" to be at the end of the line and that indentation is only one block (ie two spaces)